torchao/prototype/hqq/example.py

```diff
@@ -0,0 +1,60 @@
+import torch
+from torchao.prototype.hqq.core import HQQQuantizer
+from torchao.dtypes.affine_quantized_tensor import (
+    AffineQuantizedTensor,
+    ZeroPointDomain,
+    PlainAQTLayout,
+    PlainLayoutType,
+    TensorCoreTiledAQTLayout,
+    TensorCoreTiledLayoutType,
+    MappingType,
+)
+
+#Parameters
+device, compute_dtype = "cuda:0", torch.bfloat16
+nbits, group_size, axis = 4, 64, 1
+
+linear_layer = torch.nn.Linear(4096, 11800, bias=False, device=device)
+x = torch.randn((1, linear_layer.in_features), dtype=torch.float, device=device)/20.
+y_ref = linear_layer(x)
+W = linear_layer.weight.data.clone().to(device=device, dtype=compute_dtype)
+del linear_layer.weight 
+################################################################################################
+
+q_tensor_default = AffineQuantizedTensor.from_float(
+        input_float=W,
+        mapping_type=MappingType.ASYMMETRIC,
+        block_size=[1, group_size],
+        target_dtype=torch.uint8,
+        quant_min=0,
+        quant_max=2**nbits - 1,
+        preserve_zero=False,#Important
+        zero_point_domain= ZeroPointDomain.FLOAT,
+        layout_type=PlainLayoutType(),
+        )
+
+linear_layer.weight = q_tensor_default
+print("Default dequantization error", (W - q_tensor_default.dequantize()).abs().mean().item())
+print('Default Dot product error', (y_ref - linear_layer(x.to(compute_dtype))).abs().mean().item())
+# Default dequantization error 0.001953125
+# Default Dot product error 0.0057801781222224236
+
+
+q_tensor_hqq = AffineQuantizedTensor.from_float(
+        input_float=W,
+        mapping_type=MappingType.ASYMMETRIC,
+        block_size=[1, group_size],
+        target_dtype=torch.uint8,
+        quant_min=0,
+        quant_max=2**nbits - 1,
+        preserve_zero=False,#Important
+        zero_point_domain= ZeroPointDomain.FLOAT,
+        layout_type=PlainLayoutType(),
```

Comment:
does all these arguments also make sense for hqq? or is this list of args just partially used?

Reply:
* mapping_type is needed to be `asymmetric` because it's only for asymmetric quantization. 
* `block` size is needed  
* `quant_min` / `quant_max` are needed as well since they define the range,
*  `zero_point_domain` should be float since we can't updated the zero-point with int, 
* `layout_type` doesn't depend on the method.
 
The only thing left is `preserve_zero`, I don't really know what it is doing exactly and it's not used in hqq. But when I was comparing with the default settings, `preserve_zero=True` which is the default was giving bad results, so I set it to False. 

So in short, all the args are needed except `preserve_zero` and `layout_type` since the default args for the rest do not match with hqq logic. 
There's a check, not sure if assert is enough here, let me know how you guys do it properly in torchao:
```Python
assert zero_point_domain == ZeroPointDomain.FLOAT and mapping_type == MappingType.ASYMMETRIC and quant_min==0, "Invalid input parameters for HQQ quantization."
```

Reply:
you when you set `zero_point_domain` to float, typically preserve_zero should be False as well.